### ISSUE TITLE
[consistent-type-definitions] Interface gets changed into type when extending Array.prototype, breaking the code

### ISSUE THREAD

**Issue by Eugleo** [typescript-eslint/typescript-eslint#2707]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "@typescript-eslint/consistent-type-definitions": ["error", "type"]
}
```

```TS
declare global {
  interface Array<T> = {
    intersperse(mkT: (ix: number) => T): T[];
  };
}

if (!('intersperse' in Array.prototype)) {
  Array.prototype.intersperse = function intersperse<T>(this: T[], mkT: (ix: number) => T): T[] {
    return this.reduce((acc: T[], d, ix) => [...acc, mkT(ix), d], []).slice(1);
  };
}
```

**Expected Result**

The `interface` line should stay as it is, because the code won't properly extend `Array` if `type` is used instead.

**Actual Result**

`Interface` got autofixed into `type`, breaking the code.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.5.0` |
| `@typescript-eslint/parser`        | `4.5.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.12.0` |

**Comment by bradzacher** [comment]
Happy to accept a PR to remove the fixer when the interface is within a `declare global`
